torch/autograd/grad_mode.py

```diff
@@ -18,6 +19,12 @@ class _DecoratorContextManager:
     """Allow a context manager to be used as a decorator"""
 
     def __call__(self, func: F) -> F:
+        if inspect.isclass(func):
+            warnings.warn("Decorating classes is deprecated and will be disabled in "
+                          "future version. You should only decorate function or methods. "
```

Comment:
Do you have to specify from which version onwards? Or do we allow release notes to handle this?
```suggestion
                          "future versions. You should only decorate function or methods. "
```

Reply:
Doesn't "disabled in feature versions" imply the next version?

Reply:
Vague is fine here I think.